sdk/identity/src/token_credentials/specific_azure_credential.rs

```diff
@@ -0,0 +1,340 @@
+use super::options;
+#[cfg(not(target_arch = "wasm32"))]
+use crate::AzureCliCredential;
+#[cfg(feature = "client_certificate")]
+use crate::ClientCertificateCredential;
+use crate::{
+    AppServiceManagedIdentityCredential, ClientSecretCredential, EnvironmentCredential,
+    TokenCredentialOptions, VirtualMachineManagedIdentityCredential, WorkloadIdentityCredential,
+};
+use azure_core::{
+    auth::{AccessToken, TokenCredential},
+    error::{ErrorKind, ResultExt},
+    Error,
+};
+use std::sync::Arc;
+
+pub const AZURE_CREDENTIAL_TYPE: &str = "AZURE_CREDENTIAL_TYPE";
+
+pub mod azure_credential_types {
+    pub const ENVIRONMENT: &str = "environment";
+    #[cfg(not(target_arch = "wasm32"))]
+    pub const AZURE_CLI: &str = "azurecli";
+    pub const VIRTUAL_MACHINE: &str = "virtualmachine";
+    pub const APP_SERVICE: &str = "appservice";
+    pub const CLIENT_SECRET: &str = "clientsecret";
+    pub const WORKLOAD_IDENTITY: &str = "workloadidentity";
+    #[cfg(feature = "client_certificate")]
+    pub const CLIENT_CERTIFICATE: &str = "clientcertificate";
+}
+
+/// Creates a `DefaultAzureCredential` by default with default options.
+/// If `AZURE_CREDENTIAL_TYPE` environment variable is set, it creates a `SpecificAzureCredential` with default options.
+pub fn create_credential() -> azure_core::Result<Arc<dyn TokenCredential>> {
+    create_credential_with_options(options::TokenCredentialOptions::default())
+}
+
+fn create_credential_with_options(
+    options: TokenCredentialOptions,
+) -> azure_core::Result<Arc<dyn TokenCredential>> {
+    let env = options.env();
+    match env.var(AZURE_CREDENTIAL_TYPE) {
+        Ok(_) => SpecificAzureCredential::create(options)
+            .map(|cred| Arc::new(cred) as Arc<dyn TokenCredential>),
+        Err(_) => crate::DefaultAzureCredentialBuilder::default()
+            .with_options(options)
+            .build()
+            .map(|cred| Arc::new(cred) as Arc<dyn TokenCredential>),
+    }
+}
+
+/// Creates a new `SpecificAzureCredential` with the default options.
+pub fn create_specific_credential() -> azure_core::Result<Arc<dyn TokenCredential>> {
+    Ok(Arc::new(SpecificAzureCredential::create(
+        TokenCredentialOptions::default(),
+    )?))
+}
+
+#[derive(Debug)]
+pub enum SpecificAzureCredentialEnum {
```

Comment:
Suggest `SpecificAzureCredentialKind`. Or is an "Enum" suffix idiomatic? I have to say I've not seen an established pattern in rust and, in the face of none, we prefer "Kind" as a differentiator.

Reply:
Works for me.